docs/how_to_guide/categorical_predictors.md

```diff
@@ -0,0 +1,120 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Construct Design Matrices for Categorical Features
+
+## Design Matrix with `patsy`
```

Comment:
should be clear here that `patsy` is optional, like scikit learn, but that might be hlpeful

Reply:
I added a dropdown admonition